contrib/cirrus/setup_environment.sh

```diff
@@ -26,70 +38,57 @@ do
-        fedora-28)
-            echo ">>>>> OVERRIDING source-built runc with latest package <<<<<"
-            dnf update -y runc
-            ;&  # Continue to the next item
-        centos-7) ;&
-        rhel-7)
-            ;;
-        *) bad_os_id_ver ;;
-    esac
+# Sometimes environment setup needs to vary between distros
+# Note: This should only be used for environment variables, and temporary workarounds.
+#       Anything externally dependent, should be made fixed-in-time by adding to
+#       contrib/cirrus/packer/*_setup.sh to be incorporated into VM cache-images
+#       (see docs).
+case "${OS_RELEASE_ID}-${OS_RELEASE_VER}" in
+    ubuntu-18) ;;
+    fedora-29)
+        # Workaround for runc bug prior to
+        # https://bodhi.fedoraproject.org/updates/FEDORA-2019-6174b47003
+        # (CON_SEL may be removed after that is released)
+        CON_SEL="https://kojipkgs.fedoraproject.org/packages/container-selinux/2.100/1.git3b78187.fc29/noarch/container-selinux-2.100-1.git3b78187.fc29.noarch.rpm"
+        echo ">>>>> OVERRIDING container-selinux WITH $CON_SEL <<<<<"
+        dnf -y install $CON_SEL
+        ;;
+    fedora-28) ;;
+    centos-7) ;;
+    rhel-7) ;;
+    *) bad_os_id_ver ;;
+esac
 
-    cd "${GOSRC}/"
-    # Reload to incorporate any changes from above
-    source "$SCRIPT_BASE/lib.sh"
+cd "${GOSRC}/"
+# Reload to incorporate any changes from above
+source "$SCRIPT_BASE/lib.sh"
 
-    case "$SPECIALMODE" in
-        rootless)
-            X=$(echo "export ROOTLESS_USER='some${RANDOM}dude'" | \
-                tee -a "$HOME/$ENVLIB") && eval "$X" && echo "$X"
-            setup_rootless
-            ;;
-        in_podman)  # Assumed to be Fedora
-            dnf install -y podman buildah
-            $SCRIPT_BASE/setup_container_environment.sh
-            ;;
-    esac
-fi
+echo "Installing cni config, policy and registry config"
+req_env_var "${FUNCNAME[0]}()" "
+    GOSRC $GOSRC
+"
+sudo install -D -m 755 $GOSRC/cni/87-podman-bridge.conflist \
+                       /etc/cni/net.d/87-podman-bridge.conflist
+sudo install -D -m 755 $GOSRC/test/policy.json \
+                       /etc/containers/policy.json
+sudo install -D -m 755 $GOSRC/test/redhat_sigstore.yaml \
+                       /etc/containers/registries.d/registry.access.redhat.com.yaml
+sudo install -D -m 755 $GOSRC/test/registries.conf \
+                       /etc/containers/registries.conf
+# cri-o if installed will mess with testing in non-obvious ways
+rm -f /etc/cni/net.d/*cri*
 
-show_env_vars
+go get github.com/onsi/ginkgo/ginkgo
+go get github.com/onsi/gomega/...
 
-record_timestamp "env. setup end"
+case "$SPECIALMODE" in
+    rootless)
+        X=$(echo "export ROOTLESS_USER='some${RANDOM}dude'" | \
+            tee -a "$HOME/$ENVLIB") && eval "$X" && echo "$X"
+        setup_rootless
+        ;;
+    in_podman)  # Assumed to be Fedora
+        dnf install -y podman buildah
+        $SCRIPT_BASE/setup_container_environment.sh
+        ;;
```

Comment:
Would you consider adding a nop for `none`, and a fatal exit ("unrecognized SPECIALMODE=$SPECIALMODE") for `*`? It might help a future maintainer.

Reply:
Oh that's a good idea, thanks.